File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)

        i = n // 2
        r = n
        l = -1

        # num = nums[i]
        # if num == target:
        #     return i
        
        while i > l and i < r:
            num = nums[i]

            if num == target:
                return i
            elif num > target:
                r = i
                i = i // 2
            elif num < target:
                l = i
                i = (i + r) // 2
        return -1

File: Data Structures & Algorithms/concatenation-of-array/submission-2.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n = nums.size();
        int newLength = n*2;
        vector<int> ans;
        ans.resize(newLength);

        for (int i = 0; i < n; i++){
            ans[i] = nums[i];
            ans[n+i] = nums[i];
        }

        return ans;
    }
};